example/viam_example_app/lib/main.dart

```diff
@@ -102,22 +102,34 @@ class _MyHomePageState extends State<MyHomePage> {
     });
   }
 
-  //TODO refactor to make all _is methods the same
-
-  bool _isCamera(ResourceName rn) {
-    return rn.subtype == 'camera';
-  }
-
-  bool _isSensor(ResourceName rn) {
-    return rn.subtype == Sensor.subtype.resourceSubtype || rn.subtype == MovementSensor.subtype.resourceSubtype;
+  StreamClient _getStream(ResourceName name) {
+    return _robot.getStream(name.name);
   }
 
-  bool _isServo(ResourceName rn) {
-    return rn.subtype == Servo.subtype.resourceSubtype;
+  bool _isImplemented(ResourceName rname) {
```

Comment:
Thanks for doing this, I think this is a great way to approach it.

[nit] 
```suggestion
  bool _hasScreen(ResourceName rname) {
```

or maybe
```suggestion
  bool _isNavigable(ResourceName rname) {
```

I get what you mean with `_isImplemented` but I wonder if that makes more sense to us internally vs a consumer of the SDK. Just a [nit] take it or leave it :)